File: Share/src/main/java/com/buguagaoshu/share/domain/IpData.java
package com.buguagaoshu.share.domain;

import jakarta.persistence.*;
import lombok.Data;

/**
 * @author Pu Zhiwei {@literal puzhiweipuzhiwei@foxmail.com}
 * @create 2024-05-08
 */
@Data
@Entity
@Table(name = "ip_table")
public class IpData {
    @Id
    public String ip;

    private String createUser;

    private Long createTime;

    private Long modifyTime;

    public Integer hasEnable;
}

File: Go/internal/model/ip_data.go
package model

// IpData 对齐 Spring com.buguagaoshu.share.domain.IpData（表 ip_table，ip 为主键）
type IpData struct {
	Ip         string `gorm:"primaryKey;column:ip" json:"ip"`
	CreateUser string `gorm:"column:create_user" json:"createUser"`
	CreateTime *int64 `gorm:"column:create_time" json:"createTime"`
	ModifyTime *int64 `gorm:"column:modify_time" json:"modifyTime"`
	HasEnable  *int   `gorm:"column:has_enable" json:"hasEnable"`
}

func (IpData) TableName() string { return "ip_table" }
